workflow/controller/operator.go

```diff
@@ -1392,6 +1399,30 @@ func (woc *wfOperationCtx) executeTemplate(nodeName string, orgTmpl wfv1.Templat
 		woc.updated = true
 	}
 
+	localParams := make(map[string]string)
+	// Inject the pod name. If the pod has a retry strategy, the pod name will be changed and will be injected when it
+	// is determined
+	if resolvedTmpl.IsPodType() && resolvedTmpl.RetryStrategy == nil {
+		localParams[common.LocalVarPodName] = woc.wf.NodeID(nodeName)
+	}
+
+	// Inputs has been processed with arguments already, so pass empty arguments.
+	processedTmpl, err := common.ProcessArgs(resolvedTmpl, &args, woc.globalParams, localParams, false)
+	if err != nil {
+		return woc.initializeNodeOrMarkError(node, nodeName, templateScope, orgTmpl, opts.boundaryID, err), err
+	}
+
+	// If memoization is on, check if node output exists in cache
+	if resolvedTmpl.Memoize != nil && node == nil {
+		c := woc.controller.cache
+		storedOutput, err := c.Load(processedTmpl.Memoize.Key, processedTmpl.Memoize.Cache.ConfigMapName.Name)
+		if err != nil {
+			return nil, err
+		}
+		node = woc.initializeCacheNode(nodeName, processedTmpl, templateScope, orgTmpl, opts.boundaryID, storedOutput)
```

Comment:
We don't check if there's actually a cache entry? What if there's no error and `storedOutput` is nil (entry does not exist) then we wouldn't actually run the node at all and just initialize it with empty outputs

Reply:
Fixed by only returning the node if `storedOutput` is not nil